### ISSUE TITLE
How do m2a merge works ? (element type field should be generated once)

### ISSUE THREAD

**Issue by tigrou-ux** [B3rn475/almostjs#6]
Hello B3rn475,

I try to understand how m2a work and I miss something. The following code throws "element type field should be generated once". I'd like the obj1, 2 and 3 to be merged into entity like 
```
{
                id: "entity",
                type: "entity",
                attributes: {
                    elementnames : ["myelement1","myelement2","myelement3"]
                },
                metadata: {}
}
```

```
var extendModel = createExtender({
  type: {},
  relation: {},
  custom: {}
});

var buildModel = function() {
    var obj1 = {
        id: "partial-element-1",
        type: "partial-element",
        attributes: {
            elementname: "myelement1"
        },
        metadata: {}
    }
    var obj2 = {
        id: "partial-element-2",
        type: "partial-element",
        attributes: {
            elementname: "myelement2"
        },
        metadata: {}
    }
    var obj3 = {
        id: "partial-element-3",
        type: "partial-element",
        attributes: {
            elementname: "myelement3"
        },
        metadata:{}
    }
    return extendModel({
        elements: [obj1, obj2, obj3],
        relations: [],
        metadata: {}
    })
}
var transformer = createTransformer({
    element: [
        createRule(function(element, model) {
            return true;
        }, function(element, model) {
            var el = {
                id: "entity",
                type: "entity",
                attributes: {
                    elementnames : []
                },
                metadata: {}
            }
            el.attributes.elementnames.push(element.attributes.elementname)
            return {
                elements: [el],
                relations: [],
                metadata: {} 
            }
        })
    ]
}, 'm2a');
var model = buildModel();
console.log(util.inspect(model));
var modelTransformed = transformer(model);
```

Thank you !

**Comment by B3rn475** [comment]
The problem is that the `type` field is marked as `Single` in the `m2a` configuration.
Given this configuration the system is expecting your code to generate that field only once.

In your specific example the fix is relatively easy.
```javascript
var transformer = createTransformer({
    model: [
        createRule(function(model) {
            return true;
        }, function(model) {
            return {
                elements: [{
                    id: "entity",
                    type: "entity",
                    attributes: {
                        elementnames : []  // Not required, but here for readability
                    },
                }],
            };
        })
    ],
    element: [
        createRule(function(element, model) {
            return true;
        }, function(element, model) {
            return {
                elements: [{
                    id: "entity",
                    attributes: {
                        elementnames : [element.attributes.elementname],
                    },
                }],
            };
        }),
    ],
}, 'm2a');
```

To be honest the requirement of the `type` field, which currently is `Single` (https://github.com/B3rn475/almostjs-core/blob/master/lib/reducer.js#L39), should be relaxed to a more general `Same` which accepts multiple values, but only if they are the same.

**Comment by tigrou-ux** [comment]
Thanks! I like your project, continue!

**Comment by B3rn475** [comment]
Let me know, if it solves your problem.

If you are building something interesting with ALMOsT.js, and you would like to shared, I'm thinking about adding a "Projects using ALMOsT.js" sections in the `README.md`
